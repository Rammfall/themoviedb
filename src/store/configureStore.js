import { createLogicMiddleware } from 'redux-logic'
import { createStore, applyMiddleware } from 'redux'

import logics from './logics'
import rootReducer from './rootReducer'

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line
    const { composeWithDevTools } = require('redux-devtools-extension')

    return composeWithDevTools(applyMiddleware(...middleware))
  }

  return applyMiddleware(...middleware)
}

const configureStore = (initialState = {}) => {
  // Add dependencies to pass them to logic functions
  const dependencies = {}

  const logicMiddleware = createLogicMiddleware(logics, dependencies)
  const store = createStore(rootReducer, initialState, bindMiddleware([logicMiddleware]))

  return store
}

export default configureStore
