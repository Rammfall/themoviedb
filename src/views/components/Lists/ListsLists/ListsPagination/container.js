import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import Pagination from 'Views/components/Pagination'
import { getListsTotalSelector } from 'Store/concepts/lists/selectors'

const ListsPagination = ({ total }) => (
  <Pagination total={total} />
)

ListsPagination.propTypes = {
  total: PropTypes.number
}

ListsPagination.defaultProps = {
  total: 0
}

const mapStateToProps = (state) => ({
  total: getListsTotalSelector(state)
})

export { ListsPagination as ListsPaginationContainer }
export default connect(mapStateToProps)(ListsPagination)
