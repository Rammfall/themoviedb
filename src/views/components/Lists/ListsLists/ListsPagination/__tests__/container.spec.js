import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import diveTo from 'TestUtils/diveToEnzyme'

import ListsPagination, { ListsPaginationContainer } from '../container'

jest.mock('Store/concepts/lists/selectors', () => ({
  getListsTotalSelector: jest.fn(() => 3)
}))

describe('ListsPagination', () => {
  it('matches snapshot', () => {
    const store = configureStore()()
    const wrapper = shallow(
      <ListsPagination store={store} />
    )
    const container = diveTo(wrapper, ListsPaginationContainer)

    expect(container).toMatchSnapshot()
  })
})
