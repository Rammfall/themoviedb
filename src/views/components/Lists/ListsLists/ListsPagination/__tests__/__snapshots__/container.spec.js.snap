// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListsPagination matches snapshot 1`] = `
<Pagination
  total={3}
/>
`;
